### DISCUSSION TITLE
Enforce users joining channels & passing custom parameters to callbacks

### DISCUSSION THREAD

**Discussion by LordST95** [python-telegram-bot/python-telegram-bot#2380] [Q&A] [answered]
Hi
I'm going to write a bot, which will have two specific features, one is for statics purpose(consists of timeOfRequest,userID,commandName,totalExecutionTime) and another is a common feature know as only give service to users who are joined to a special channel.

the naive approach is at the beginning of each command handler function ( for example [echo  func](https://github.com/python-telegram-bot/python-telegram-bot/blob/eee892159833678ef56db6201434c5ae7fa8fd48/examples/echobot.py#L44) ), I check the subscription of the user and initial the staticData then at the end of that function, I update that staticData and store it on my DB.

but

the better approach is to extract those two features and do them when a new update received and add them to context of command handler function. as I read your source code, I should somehow override [Dispatcher](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.dispatcher.html#telegram-ext-dispatcher) and add my objects to its [CallBackContext](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.callbackcontext.html#telegram-ext-callbackcontext) but I can't figure out how should I do this in a very good programmatically way ...

Can someone help me?

<br>
<br>
my other question is, how it is possible to do sth like bellow:

```
some_handler = CommandHandler('some', some)
    dispatcher.add_handler(some_handler, anotherParameter=obj)     # pass object to handler when registering it
```
and access it in handler function:
```
def some(update, context, *args, **kwargs):
    myObj = kwargs.get('anotherParameter')     # access object

```

PTB-version: 13.1

**Comment by Bibo-Joshi** [comment]
Hi. Please see [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#how-do-i-enforce-users-joining-a-specific-channel-before-using-my-bot) and [here](https://t.me/pythontelegrambotgroup/362834).

**Comment by LordST95** [comment, marked as answer]
Thanks to @Bibo-Joshi, I could write answers as follow ...

1- for the last question, we could wrap our handlerFunction with functools as bellow:
```
from functools import partial

def start(update, context, foo=None):
    print("foo: ", foo)

start_handler = CommandHandler('start', partial(start,foo='sina'))
dispatcher.add_handler(start_handler)
```

2- for checking the subscription of users and write code for doing entrance stuff, we could write typeHandler. plz keep in mind that in this approach the group related to typeHandler should be lower than otherHandlers, because it should run before others ;)

```
def checkSubscription(update, context):
    #  check subscription with get_chat_member() or add sth to context
    pass

dispatcher.add_handler(TypeHandler(Update, callback=checkSubscription)) # default group value is 0
dispatcher.add_handler(start_handler, group=1)
dispatcher.add_handler(help_handler, group=1)
```


for more info, check out these [links](https://github.com/python-telegram-bot/python-telegram-bot/discussions/2380#discussioncomment-356393)
